src/sysfiles.c

```diff
@@ -3701,6 +3823,65 @@ static StructGVarFunc GVarFuncs [] = {
 *F * * * * * * * * * * * * * initialize module * * * * * * * * * * * * * * *
 */
 
+// This function is called by 'InitSystem', before the usual module
+// initialization.
+void InitSysFiles(void)
+{
+    memset(syBuffers, 0, sizeof(syBuf));
+
+    memset(syBuf, 0, sizeof(syBuf));
+    for (Int i = 0; i < ARRAY_SIZE(syBuf); i++)
+        syBuf[i].type = unused_socket;
```

Comment:
If we need fp=-1 when we set unused, do we need it here too?

Reply:
I am not sure we "need" `fp=-1`, but some code seemed to check for it w/o checking whether the `suBuf` entry was in use. But I agree that we should at least be consistent. To that end, it might be best to just call `SyBufMarkUnused` in this loop - it'll be slightly less efficient, but that's totally irrelevant here, and is totally worth the automatic consistency in behaviour.

Reply:
I just had a closer look, and that lead me to discover several places in our code which confuses OS file descriptors with an index into `syBuf` :-/. I'll submit an update to this PR later.

Reply:
I think defensive programming is a good idea (setting fp=-1), as we don't want code accidentally accessing 0!

Also, I did try to find places where we mishandle OS file descriptors, I hope I didn't introduce any new ones, sorry if I did.